### DISCUSSION TITLE
Dispatching Events to Browser Isn't Working

### DISCUSSION THREAD

**Discussion by NBA707** [livewire/livewire#8646] [1. Help]
I have a livewire component that has a button linked to a function in the component.  The component does a database update and dispatches an event.   I tried $wire but it said it was undefined as I am not using alpine in the blade file.

Any ideas on a simple way to get events working?

`wire:click="toggleLink({{ $row->VQD_AUTO_KEY }});"`

In the function I have:

`public function toggleLink( $vqd_auto_key )
    {
       $event = 'vqd-unlinked';
       // Some code that does some very cool stuff
        $this->dispatch($event, vqd: $vqd_auto_key);
    }`

When I click the button in the response:

`{
    "components": [
        {
            "snapshot": "... BLAH BLAH BLAH ....",
                "dispatches": [
                    {
                        "name": "vqd-unlinked",
                        "params": {
                            "vqd": 759983
                        }
                    }
                ]
            }
        }
    ],
    "assets": []
}`

In the main blade.php file I have: 

`<script>
        document.addEventListener('livewire:load', function () {
            Livewire.on('vqd-linked', (event) => {
                console.log('vqd-linked',event); //et refreshPosts = event.detail.refreshPosts
                alert('Linked')
                // ...
            });

            $Livewire.on('vqd-unlinked', (event) => {
                console.log('vqd-unlinked',event); //et refreshPosts = event.detail.refreshPosts
                alert('unLinked')
                // ...
            });
        });
    </script>`

Livewire: 3.5.2
Laravel: v10.48.15
PHP: 8.3.9

**Comment by lrljoe** [comment]
So Livewire is undefined?  Then Livewire isn't present, and you can't expect Livewire to magically inject itself into something else.

Your javascript should look like:
```
    document.addEventListener('livewire:init', () => {
       Livewire.on('vqd-linked', (event) => {
          console.log('vqd-linked',event); 
          alert('Linked')
       });
    });
```

This must be on a page, where Alpine/Livewire will load, you cannot expect a traditional Laravel Controller (for example) to understand this.

If this is living inside of your component, then I'd ask **why** you're trying to listen client-side, from the same component, that is not logical at all, as you have full control of the frontend from the backend.
